### PR TITLE
add yolov6 c++ and yolov6 pybind

### PR DESCRIPTION
**Pull request by DefTruth**
* add yolov6 c++ and yolov6 pybind
* add examples (contains yolov5 and yolov6 now) which can build with fastdeploy
* add model_zoo yolov6 c++/python demo
* fixed CMakeLists.txt typos

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +38,9 @@ option(ENABLE_VISION_VISUALIZE "if to enable visualize vision model result toolb
 option(ENABLE_OPENCV_CUDA "if to enable opencv with cuda, this will allow process image with GPU." OFF)
 option(ENABLE_DEBUG "if to enable print debug information, this may reduce performance." OFF)
 
+# Wheter to build fastdeply with vision/text/... examples, only for testings.
```
Comment (jiangjiajun) [inline CMakeLists.txt]: Whether

**Review by jiangjiajun — COMMENTED** [review]
参照`model_zoo/vision/ultralytics/yolov5`, 加上python/cpp，以及文档说明

File: CMakeLists.txt (hunk 1 above)
Comment (DefTruth) [reply CMakeLists.txt]: 已经修改拼写错误

**Comment by DefTruth** [comment]
> 
已经增加 model_zoo/vision/yolov6 python/cpp 以及文档说明

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**